modules/pricer/src/main/java/com/opengamma/strata/pricer/fxopt/SmileDeltaParameters.java

```diff
@@ -92,28 +123,68 @@ public static SmileDeltaParameters of(
       DoubleArray riskReversal,
       DoubleArray strangle) {
 
+    return of(expiry, atmVolatility, delta, riskReversal, strangle, createParameterMetadata(expiry, delta));
+  }
+
+  /**
+   * Obtains an instance from market data at-the-money, delta, risk-reversal and strangle.
+   * 
+   * @param expiry  the time to expiry associated to the data
+   * @param atmVolatility  the at-the-money volatility
+   * @param delta  the delta of the different data points, must be positive and sorted in ascending order,
+   *   the put will have as delta the opposite of the numbers
+   * @param riskReversal  the risk reversal volatility figures, in the same order as the delta
+   * @param strangle  the strangle volatility figures, in the same order as the delta
+   * @param parameterMetadata  the parameter metadata
+   * @return the smile definition
+   */
+  public static SmileDeltaParameters of(
+      double expiry,
+      double atmVolatility,
+      DoubleArray delta,
+      DoubleArray riskReversal,
+      DoubleArray strangle,
+      List<ParameterMetadata> parameterMetadata) {
+
     ArgChecker.notNull(delta, "delta");
     ArgChecker.notNull(riskReversal, "riskReversal");
     ArgChecker.notNull(strangle, "strangle");
-    ArgChecker.isTrue(delta.size() == riskReversal.size(),
+    int nbDelta = delta.size();
+    ArgChecker.isTrue(nbDelta == riskReversal.size(),
         "Length of delta {} should be equal to length of riskReversal {}", delta.size(), riskReversal.size());
-    ArgChecker.isTrue(delta.size() == strangle.size(),
+    ArgChecker.isTrue(nbDelta == strangle.size(),
         "Length of delta {} should be equal to length of strangle {} ", delta.size(), strangle.size());
-    int nbDelta = delta.size();
+
     double[] volatility = new double[2 * nbDelta + 1];
     volatility[nbDelta] = atmVolatility;
     for (int i = 0; i < nbDelta; i++) {
       volatility[i] = strangle.get(i) + atmVolatility - riskReversal.get(i) / 2.0; // Put
       volatility[2 * nbDelta - i] = strangle.get(i) + atmVolatility + riskReversal.get(i) / 2.0; // Call
     }
-    return new SmileDeltaParameters(expiry, delta, DoubleArray.ofUnsafe(volatility));
+    return of(expiry, delta, DoubleArray.ofUnsafe(volatility), parameterMetadata);
+  }
+
+  //-------------------------------------------------------------------------
+  private static ImmutableList<ParameterMetadata> createParameterMetadata(double expiry, DoubleArray delta) {
```

Comment:
This needs to be called from an `@ImmutablePreBuild` as well for compatibility.

Reply:
Added @ImmutablePreBuild